lib/runtime/wasmtime/instance.go

```diff
@@ -96,7 +96,14 @@ func newLegacyInstanceFromModule(module *wasmtime.Module, engine *wasmtime.Engin
 	}
 
 	store := wasmtime.NewStore(engine)
-	instance, err := wasmtime.NewInstance(store, module, cfg.Imports(store))
+	linker := wasmtime.NewLinker(store)
+
+	err := ImportLegacyRuntimeLinker(store, linker)
+	if err != nil {
+		return nil, err
+	}
+
+	instance, err := linker.Instantiate(module)
```

Comment:
could you update this so that `ImportLegacyRuntimeLinker` accepts the store only, and returns the linker? so you do `linker := wasmtime.NewLinker(store)` in the import function instead?